extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcIdentityProvider.java

```diff
@@ -197,14 +197,15 @@ public Uni<SecurityIdentity> apply(TokenVerificationResult result, Throwable t)
 
     private Uni<TokenVerificationResult> verifyPrimaryTokenUni(Map<String, Object> requestData,
             TokenAuthenticationRequest request, TenantConfigContext resolvedContext, UserInfo userInfo) {
+        final Uni<TokenVerificationResult> tokenVerificationResult;
         if (isInternalIdToken(request)) {
             if (requestData.get(NEW_AUTHENTICATION) == Boolean.TRUE) {
                 // No need to verify it in this case as 'CodeAuthenticationMechanism' has just created it
-                return Uni.createFrom()
+                tokenVerificationResult = Uni.createFrom()
```

Comment:
We should likely leave the internal id token code untouched - Quarkus generates an internal token itself when it deals with GitHub, etc, so applying any policies to it does not quite work.

Reply:
I think this suggestion is a setback, because it is not about _being reasonable to expect some acr values_, but about _ensuring that endpoint XYZ is only accessed with token that has specified acr values_. If we add exceptions when the token doesn't need to have `acr` values, how users can trust us that we ensure the endpoint is only access with required `acr` values? If someone requires ID token to have acrs that can't be there, that is not a security issue, but if some required access to endpoint only from tokens with specified `acr` values and we don't enforce that, aren't we introducing a security risk?

Reply:
Or fine, don't support ID token scenarios, but at least we can throw illegal state? I am worried to just ignore a check that user explicitly required.

Reply:
Hi Michal, @michalvavrik, sure, the illegal state exception can do, or shall we just break the build if `id-token-required=false` in case of the `web-app` ?

Reply:
I have added a warning when there is combination of the `id-token-required=false` and `web-app`. We cannot know when the tenant is created if it will use step up authentication and there can be multiple tenants (like in `oidc-tenancy` module) and some may require `@AuthenticationContext`.